File: Daily/PY/Leetcode187-重复的DNA序列.py
from typing import List
from collections import defaultdict

class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        n = len(s)
        res = []
        mp = defaultdict(int)
        if n < 10:
            return res
        for i in range(n - 9):
            cur = s[i:i + 10]
            mp[cur] += 1
            if mp[cur] == 2:
                res.append(cur)
        return res

File: Leetcode/sbgo/Leetcode187-重复的DNA序列.go
package sbgo

func findRepeatedDnaSequences(s string) []string {
	n := len(s)
	ans := []string{}
	if n < 10 {
		return ans
	}
	mp := map[string]int{}
	for i := 0; i <= n-10; i++ {
		cur := s[i : i+10]
		mp[cur]++
		if mp[cur] == 2 {
			ans = append(ans, cur)
		}
	}
	return ans
}
